app/controllers/shopping_lists_controller.rb

```diff
@@ -0,0 +1,32 @@
+class ShoppingListsController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy generate_shopping_list]
+
+  def index
+    general_shopping_list
+    @total_count = @shopping_list_items.sum { |item| item.quantity.to_i }
+    @total_price = @shopping_list_items.sum { |item| item.quantity.to_i * item.price }
+  end
```

Comment:
- [ ] Please kindly add this page to the nav items as stated in the requirement and make sure that the route is **`/generating_shopping_list`**. ℹ️ 

![20220419_123236](https://github.com/MarcoDDM/recipeapp/assets/103386183/6912a2d7-5ed4-454f-9ea4-70614338ab22)